Pcap++/header/PcapLiveDevice.h

```diff
@@ -130,6 +137,8 @@ namespace pcpp
 		static void onPacketArrivesNoCallback(uint8_t* user, const struct pcap_pkthdr* pkthdr, const uint8_t* packet);
 		static void onPacketArrivesBlockingMode(uint8_t* user, const struct pcap_pkthdr* pkthdr, const uint8_t* packet);
 	public:
+		PcapLiveDevice(const PcapLiveDevice& other) = delete;
```

Comment:
please also implement moving construction/assignment.

Reply:
btw, if the underlying are shared pointers, why do we disallow coping?

Reply:
> btw, if the underlying are shared pointers, why do we disallow coping?

IDK. I think the reason a `LiveDevice` itself is not copiable is something with the underlying Pcap descriptors. Although there is still the `clone` method so, not really sure. The current change is mostly a stylistic from the pre-Cpp11 style of private ctors to post-Cpp11 deleted ctors.

Reply:
we can keep the old `clone`, but I think we can now embrace the copy constructor/assignment?

Reply:
Ah, figured out at least partly why its private / deleted. `LiveDevice` is a virtual class and usually handled by pointers to base class (i.e. `WinPcapLive` device is used through `PcapLiveDevice` pointer). A copy ctor used from a base pointer will only use the base copy ctor, not the actual copy ctor of the class the base ptr points to.

Reply:
> please also implement moving construction/assignment.

Added 4bcb50e